gcp_variant_transforms/bq_to_vcf.py

```diff
@@ -135,7 +139,30 @@ def _bigquery_to_vcf_shards(
          beam.Map(_pair_variant_with_key, known_args.number_of_bases_per_shard)
          | 'GroupVariantsByKey' >> beam.GroupByKey()
          | beam.ParDo(_get_file_path_and_sorted_variants, vcf_data_temp_folder)
-         | vcfio.WriteVcfDataLines())
+         | vcfio.WriteVcfDataLines(annotation_fields))
+
+
+def _extract_annotation_fields(schema):
```

Comment:
Actually, I think you don't need to precalculate this. For any variant, you can treat any record under alternate_bases as annotation, which can be done entirely within the BigQueryToVariant transform (i.e. just looking at the json record).

Reply:
If I understand correctly, the record we got in `BigQueryToVariant` is a dictionary, which does not preserve the original order anymore.The extraction of annotation fields here is mainly for annotation fields order recovering (e.g., `Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|`).

Reply:
@allieychen Ask yourself when and if order is important.  It is good to think abstractly of what your are analyzing or processing.  Do you have different Transformer patterns - ideally one per sub-domain - that will do the work for you on the fly?  The answer should be yes, so as to write as little code as possible, and keep reusing these modular, functional components in different compositions.

Reply:
@allieychen: I see. You're right. I think in this case we should keep the order as otherwise it may be difficult to merge/compare values from multiple VCF files. Please add a comment about why it's needed though.

Reply:
@arostamianfar  A merge requires a subset of schema equivalency.  A schema is different than data.  A comparison is an intersection of schema filtered data.  Order is part of the projection operator and not really a requirement in your input data, and joins have those as defaults.  Making that your comparison and merge strategy is the more optimal approach as you scale up.